apps/src/music/progress/ProgressManager.ts

```diff
@@ -103,9 +103,11 @@ export default class ProgressManager {
         // Ask the lab-specific validator if this validation's
         // conditions are met.
         if (this.validator.conditionsMet(validation.conditions)) {
-          this.currentProgressState.satisfied = validation.next;
-          this.currentProgressState.message = validation.message;
-          this.onProgressChange();
+          if (!this.currentProgressState.satisfied) {
```

Comment:
what is the reason for this change?

Reply:
Oh, we were calling `this.onProgressChange()` repeatedly once the criteria were satisfied.  This change ensures we only call it once, the first time they are satisfied.